service/history/nDCStateRebuilder.go

```diff
@@ -180,6 +187,8 @@ func (r *nDCStateRebuilderImpl) rebuild(
 		return nil, 0, err
 	}
 
+	rebuiltMutableState.GetExecutionInfo().LastFirstEventTxnId = lastTxnId
```

Comment:
we need a setter method on mutable state to set this

Reply:
the last txn ID should be managed by history builder, not here

Reply:
Synced with Wenquan:
* use current approach overall for this PR for quick fix and unblocking of GetWorkflowExecutionHistoryReverse feature
* followup with change to move managing of transaction ids to history_builder. Which would effectively include:
  * Pass in event batch metadata to history_builder/rebuilder/etc when we apply/replay events
  * Move transaction id generation out of mutable stage
  * Make history_builder responsible for managing transaction ids